liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -1250,15 +1250,17 @@ protected void doExecute() throws Exception {
 
         processContainerParams();
 
-        if (!container) {
-            if (serverDirectory.exists()) {
-                if (ServerStatusUtil.isServerRunning(installDirectory, super.outputDirectory, serverName)) {
+        if (serverDirectory.exists()) {
+            if (ServerStatusUtil.isServerRunning(installDirectory, super.outputDirectory, serverName)) {
+                if (!container) {
                     throw new MojoExecutionException("The server " + serverName
-                            + " is already running. Terminate all instances of the server before starting dev mode."
-                            + " You can stop a server instance with the command 'mvn liberty:stop'.");
+                        + " is already running. Terminate all instances of the server before starting dev mode."
+                        + " You can stop a server instance with the command 'mvn liberty:stop'.");
+                } else {
+                    log.warn("Running server detected from previous execution that did not stop cleanly. If startup does not complete and times out do a clean before re-running.");
```

Comment:
I have two comments/questions about this warning:

1. I don't see how we can be certain that we have hit the scenario where the previous execution did not stop cleanly. It is possible that a server could just be left running (wasn't attempted to be stopped at all or failed to stop for some other reason).
2. If we are in a scenario where the server is still running, don't we still need to run the command `mvn liberty:stop`? It does not seem like a `mvn clean` will resolve the situation.

Reply:
OK. and another point:  it is weird and probably not what you want, but not technically impossible to have the server running and devc running at the same time.

How about:

"Running server detected, which could cause unexpected results.  Note this could also occur because a previous server execution did not stop cleanly.  If devc fails in this latter case, you may need to a clean before re-running."

Reply:
Yes, we had considered the scenario where devc and dev mode are running at the same time and decided it was best to avoid it, which is part of why we opened https://github.com/OpenLiberty/ci.maven/issues/842. For now, I don't think there is an issue with just showing the warning and letting it go.

I like the message modification... I'm still thinking we need to have the `mvn liberty:stop` command in there somewhere though... Maybe:

"Running server detected, which could cause unexpected results. If you would like to terminate the local running server, run the command `mvn liberty:stop`. Note this warning may also occur because a previous server execution did not stop cleanly. If devc fails in this latter case, you may need to run the command `mvn clean` before re-running."

Reply:
OK reworded with a new commit.